econtext/src/main/java/com/adyen/checkout/econtext/EContextComponent.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ozgur on 7/6/2022.
+ */
+
+package com.adyen.checkout.econtext
+
+import androidx.annotation.RestrictTo
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.adyen.checkout.action.ActionHandlingComponent
+import com.adyen.checkout.action.DefaultActionHandlingComponent
+import com.adyen.checkout.action.GenericActionDelegate
+import com.adyen.checkout.components.ButtonComponent
+import com.adyen.checkout.components.PaymentComponent
+import com.adyen.checkout.components.PaymentComponentEvent
+import com.adyen.checkout.components.PaymentComponentState
+import com.adyen.checkout.components.base.ComponentDelegate
+import com.adyen.checkout.components.base.ComponentEventHandler
+import com.adyen.checkout.components.extensions.mergeViewFlows
+import com.adyen.checkout.components.model.payments.request.EContextPaymentMethod
+import com.adyen.checkout.components.toActionCallback
+import com.adyen.checkout.components.ui.ButtonDelegate
+import com.adyen.checkout.components.ui.ViewableComponent
+import com.adyen.checkout.components.ui.view.ComponentViewType
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import kotlinx.coroutines.flow.Flow
+
+abstract class EContextComponent<EContextPaymentMethodT : EContextPaymentMethod> protected constructor(
```

Comment:
Can we add tests from this component? We have similar examples in `IssuerListComponentTest` and `OnlineBankingComponentTest`